pkg/log/ansHook.go

```diff
@@ -0,0 +1,109 @@
+package log
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/SAP/jenkins-library/pkg/ans"
+	"github.com/SAP/jenkins-library/pkg/xsuaa"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"strings"
+)
+
+// ANSHook is used to set the hook features for the logrus hook
+type ANSHook struct {
+	client ans.Client
+	event  ans.Event
+}
+
+// NewANSHook creates a new ANS hook for logrus
+func NewANSHook(config ans.Configuration, correlationID string) ANSHook {
+	ansServiceKey, err := ans.UnmarshallServiceKeyJSON(config.ServiceKey)
```

Comment:
the unmarshall does not check the content. despite a wrong key the hook is setup.

Reply:
Should we fire a test request to the service with this key to verify that it is setup correctly? Or should we check that the fields exist as they should?
Only the former makes is a real test of the validity of the service key, IMO.

Reply:
Done with `CheckCorrectSetup` function. Please review.